### PR TITLE
created activemq publisher via AMQP

### PR DESCRIPTION
**Pull request by Hennie97**

### CODE REVIEW

File: activemq/publisherAMQP.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+func NewAMQPPublisher(request NewAMQPPublisherRequest) AMQPPublisher {
+	return AMQPPublisher{
+
+	}
+}
+
+type AMQPPublisher struct {
+	Client *amqp.Client
```
Comment (bouwerp) [inline activemq/publisherAMQP.go]: add host and port, and credentials username and password (but make all private):
```suggestion
	client *amqp.Client
	conn net.Conn
	host string
	Port int
	username string
	password string
```

File: activemq/publisherAMQP.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import (
+	"context"
+	"github.com/andile-innovation/popcorn/log"
+	"net"
+	"pack.ag/amqp"
+	"time"
+)
+
+type NewAMQPPublisherRequest struct {
+	Host string
```
Comment (bouwerp) [inline activemq/publisherAMQP.go:13]: Include credentials here:
```suggestion
	Host string
	Username string
	Password string
```

File: activemq/publisherAMQP.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+	}
+}
+
+type AMQPPublisher struct {
+	Client *amqp.Client
+	Conn net.Conn
+}
+
+func (p *AMQPPublisher) Connect(address string) error {
```
Comment (bouwerp) [inline activemq/publisherAMQP.go]: Don't pass address to connect - rater utilise constructor parameters `p.host` and `p.port`

File: activemq/publisherAMQP.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	)
+	if err != nil {
+		return err
+	}
+
+	p.Client = client
+	return nil
+}
+
+func (p *AMQPPublisher) Publish(destination string, data []byte) error {
```
Comment (bouwerp) [inline activemq/publisherAMQP.go]: keep arguments the same as the interface:
```suggestion
func (p *AMQPPublisher) Publish(channel string, data []byte) error {
```

File: activemq/publisherAMQP.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		}
+		defer func() {
+			err := sender.Close(ctx)
+			if err != nil {
+				log.Error(err)
+			}
+		}()
+
+		ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+
```
Comment (bouwerp) [inline activemq/publisherAMQP.go]:
```suggestion
defer cancel()
```

File: activemq/publisherAMQP.go [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+		// Send message
+		err = sender.Send(ctx, amqp.NewMessage(data))
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+
+		cancel()
```
Comment (bouwerp) [inline activemq/publisherAMQP.go]:
```suggestion
```

**Review by bouwerp — CHANGES_REQUESTED** [review]

File: activemq/publisherAMQP.go (hunk 3 above)
Comment (bouwerp) [reply activemq/publisherAMQP.go]: @Hennie97 did you resolve this? I see it's not yet pushed

Comment (Hennie97) [reply activemq/publisherAMQP.go]: I resolved it and pushed it now.

File: activemq/subscriberAMQP.go [hunk 7]
```diff
@@ -0,0 +1,97 @@
+package activemq
```
Comment (bouwerp) [inline activemq/subscriberAMQP.go]: this code should not be on this branch - rather separate

**Review by bouwerp — CHANGES_REQUESTED** [review]

**Review by bouwerp — APPROVED** [review]

**Pull request merged**